File: sources/static/js/tweet.js
var Tweet = {
	rt:'',
	twitter_user:'',
	is_rt:false,
	texto:'',
	fav:'',
	media_url:'',

	contructorFromJSON:function(tweet){
		Tweet.rt = tweet.rt;
		Tweet.twitter_user = tweet.tuser;
		if(tweet.is_rt == "False"){
			Tweet.is_rt = false;
		}else{
			Tweet.is_rt = true;
		}
		Tweet.texto = tweet.text;
		Tweet.fav = tweet.fav;

		Tweet.media_urls = tweet.media;
		console.log(Tweet.media_urls);
		if(Tweet.media_urls != '' && Tweet.media_urls != 'None'){

			Tweet.media_urls = JSON.parse(Tweet.media_urls);
		}else{
			Tweet.media_urls = '';
		}
		Tweet.user_img = tweet.user_img;
	},

	toHTML:function(border){
		//var cadena = '<div class="delimitador"></div>';
		var cadena = '';
		cadena += Tweet.relevanciaToHTML(border);
		cadena += Tweet.usrImgToHTML();
		cadena += '<div class="cont-tweet">';
		cadena += Tweet.tweetUsrToHTML();
		cadena += '<div class="tweet-text">';
		cadena += Tweet.textoToHTML();
		cadena += '</div>';
		cadena += '<div class="contenedor-rf">';
		cadena += '<span class="fav-count">';
		cadena += Tweet.fav;
		cadena += '</span>';
		cadena += '<span class="retweet-count">';
		cadena += Tweet.rt;
		cadena += '</span>';
		cadena += '</div>';
		//#tiene algún tipo de objeto multimedia

		if (Tweet.media_urls != ''){
			cadena += Tweet.multimediaToHTML();
		}
		cadena += '</div>';

		

		cadena += '</div>';
		return cadena;
	},

	relevanciaToHTML : function(border){
		if (border == true)
			return '<div class="tweet border-tweet">';
		else
			return '<div class="tweet">';
	},
	usrImgToHTML:function(){
		var cadena = '<div class="cont-user-img">';
		cadena += '<img src="'+Tweet.user_img+'">';
		cadena += '</div>';
		return cadena;
	},
	tweetUsrToHTML :function(){
		var cadena = '<div class="cont-usr">';
		cadena += '<a href="https://twitter.com/'+Tweet.twitter_user+'" target="_blank">@'+Tweet.twitter_user+ '</a>';
		cadena += '</div>';
		return cadena;
	},

	textoToHTML : function(){
		var palabras = Tweet.texto.split(" ");
		var cadena = '';

		var tam = palabras.length;
		var palabra;
		for (var i = 0; i < tam; i++) {
			palabra = palabras[i];

			if((palabra.indexOf("http://") > -1) || (palabra.indexOf("https://") > -1)){
				cadena += '<a class="link" href="'+ palabra+'" target="_blank">'+palabra+'</a> ';
			}else if(palabra.indexOf("@") > -1){
				var user = Tweet.limpiaTwitterUser(palabra);
				cadena += '<a class="user" href="https://twitter.com/'+ user +'" target="_blank">'+palabra+'</a> ';
			}else{
				cadena += palabra + " ";
			}
			
		}
		return cadena;
	},

	multimediaToHTML:function(){

		var cadena = '';
		for (var i = 1; i <= 4; i++) {
			
			var key = 'photo_'+i;
            //alert("Mine is " + i + "|" + item.title + "|" + item.key);
            if(key in Tweet.media_urls){
				cadena += '<div class="cont-multi">';
				cadena += '<a href="'+ Tweet.media_urls[key] +'" target="_blank">';
				cadena += '<div class="img_busqueda" style="background-image: url(\'';
				//#http://pbs.twimg.com/media/BziK-yaIIAA1uW2.jpg
				cadena += Tweet.media_urls[key];
				cadena += '\');">';
			
				cadena += '</div>';
				cadena += '</a>';
				cadena += '</div>';
			}else{
				break;
			}
        };

		return cadena;
	},
	limpiaTwitterUser : function(user){
		var cadena = '';

		for (var i = user.length - 1; i >= 0; i--) {
			caracter = user[i];
			if(caracter != '@' && caracter != ':'){
				cadena = caracter + cadena;
			}
		};

		return cadena;
	}
};

File: sources/Web/Tweet.py
# -*- coding: utf-8 -*-
class Tweet(object):
	"""docstring for Tweet"""
	@staticmethod
	def imprimeTweett(tweet, relevancia=True):
		palabras = tweet[0].split(" ")
		cadena = '<div class="delimitador"></div>'
		if relevancia == True:
			cadena += Tweet.imprimeTweettRelevancia(tweet)
		else:
			cadena += '<div class="tweet">'

		cadena += Tweet.imprimeTweettUsrIMG()

		cadena += '<div class="cont-tweet">'
		cadena += Tweet.imprimeTweetUsr(tweet)
		cadena += '<div class="tweet-text">'
		for palabra in palabras:
			#es un link
			if "http://" in palabra or "https://" in palabra:
				cadena += '<a class="link" href="'+ palabra+'" target="_blank">'+palabra+'</a> '
			#es un usuario de twitter
			elif "@" in palabra:
				user = Tweet.limpiaTwitterUser(palabra)
				cadena += '<a class="user" href="https://twitter.com/'+ user +'" target="_blank">'+palabra+'</a> '
			#cualquier otra cosa			
			else:
				cadena += palabra + " "

		cadena += '</div>'
		cadena += '<div class="contenedor-rf">'
		cadena += '<span class="fav-count">'
		cadena += str(tweet[1])
		cadena += '</span>'
		cadena += '<span class="retweet-count">'
		cadena += str(tweet[2])
		cadena += '</span>'
		cadena += '</div>'
		#tiene algún tipo de objeto multimedia
		if tweet[4] != '':
			cadena += Tweet.imprimeTweettMedia(tweet[4])
			
		cadena += '</div>'

		

		cadena += '</div>'
		return cadena
	@staticmethod
	def imprimeTweettUsrIMG():
		cadena = '<div class="cont-user-img">'
		cadena += '<img src="static/img/tweetuser.png">'
		cadena += '</div>'
		return cadena

	@staticmethod
	def imprimeTweetUsr(tweet):
		cadena = '<div class="cont-usr">'
		cadena += '<a href="https://twitter.com/'+tweet[5]+'" target="_blank">@'+tweet[5]+ '</a>'
		cadena += '</div>'
		return cadena

	@staticmethod
	def imprimeTweettRelevancia(tweet):
		imOrRt = ""

		if tweet[3] == True:
			imOrRt = "rt"
		else:
			imOrRt = "im"

		relevancia = (float(tweet[1]) * 0.75) + (float(tweet[2]) * 1.5)
		if relevancia == 0:
			return '<div class="tweet '+imOrRt+'1">'
		elif relevancia < 100:
			return '<div class="tweet '+imOrRt+'2">'
		elif relevancia < 750:
			return '<div class="tweet '+imOrRt+'3">'
		elif relevancia < 5000:
			return '<div class="tweet '+imOrRt+'4">'
		else:
			return '<div class="tweet '+imOrRt+'5">'

	@staticmethod
	def imprimeTweettMedia(media):
		cadena = ''

		if '.jpg' in media or '.png' in media:
			cadena += '<div class="cont-multi">'
			cadena += '<a href="'+ media +'" target="_blank">'
			cadena += '<div class="img_busqueda" style="background-image: url(\''
			#http://pbs.twimg.com/media/BziK-yaIIAA1uW2.jpg
			cadena += media 
			cadena += '\');">'
		
			cadena += '</div>'
			cadena += '</a>'
			cadena += '</div>'

		return cadena

	@staticmethod
	def limpiaTwitterUser(user):
		cadena = ''

		for caracter in user:
			if caracter != '@' and caracter != ':':
				cadena += caracter

		return cadena
		